### ISSUE TITLE
Server does not remove unset optional task args values

### ISSUE THREAD

**Issue by rkpasia** [fractal-analytics-platform/fractal-web#196]
With the changes required by #190, the server will no longer receive payloads with properties that have `null` values.

This will bing to the following scenario:

Given the following schema:
```json
{
  "additionalProperties": false,
  "properties": {
    "sleep_time": {
      "description": "Interval to sleep, in seconds.",
      "title": "Sleep Time",
      "type": "integer"
    }
  },
  "required": [],
  "title": "DummyParallel",
  "type": "object"
}
```

If an user sets, through the client by saving edits, the `sleep_time` property, then the server will receive the following HTTP request, with given payload.

```HTTP
PATCH /api/v1/project/1/workflow/2/wftask/4
{
    "args": {
        "raise_error": true,
        "message": "test",
        "sleep_time": 1
    }
}
```

Upon following refresh of the page, the client will receive the same payload, saved before.

*Let's assume that the user wants to remove the optional parameter `sleep_time`*

To do so, the user deletes from the UI the number inside the form and confirms to save changes.

The fractal server, after #190 will receive the following payload.

```HTTP
PATCH /api/v1/project/1/workflow/2/wftask/4
{
    "args": {
        "raise_error": true,
        "message": "test"
    }
}
```

The user now expects that, upon refreshing the page, *or upon running the task*, the `sleep_time` property is not set.
Instead, the server, will handle the client the following payload.

```HTTP
GET /api/v1/project/1/workflow/2
[...]
    "args": {
        "raise_error": true,
        "message": "test",
        "sleep_time": 1
    }
[...]
```

Shall we expect that the server deletes or sets the `sleep_time` property to `null` or deletes it? @tcompa @jluethi

**Comment by tcompa** [comment]
> Shall we expect that the server deletes or sets the sleep_time property to null or deletes it? 

If we move on as in https://github.com/fractal-analytics-platform/fractal-server/issues/758, then we should expect that the server deletes `args["sleep_time"]`, when the patch-workflowtask endpoin receives a payload like
```json
{
    "args": {
        "raise_error": true,
        "message": "test"
    }
}
```

**Comment by tcompa** [comment]
> > Shall we expect that the server deletes or sets the sleep_time property to null or deletes it?
> 
> If we move on as in [fractal-analytics-platform/fractal-server#758](https://github.com/fractal-analytics-platform/fractal-server/issues/758), then we should expect that the server deletes `args["sleep_time"]`, when the patch-workflowtask endpoin receives a payload like
> 
> ```json
> {
>     "args": {
>         "raise_error": true,
>         "message": "test"
>     }
> }
> ```

We now released fractal-server 1.3.0a9, and this is the correct behavior to expect.

Note that for parameters that do have a default value in the args schema (which is not the case for `sleep_time`, in this case), removing them corresponds to resetting their value to the default.

**Comment by tcompa** [comment]
Closed as of recent fractal-server update
